assert/assertion_order.go

```diff
@@ -286,6 +235,11 @@ func Less(t TestingT, e1 interface{}, e2 interface{}, msgAndArgs ...interface{})
 //    assert.LessOrEqual(t, "a", "b")
 //    assert.LessOrEqual(t, "b", "b")
 func LessOrEqual(t TestingT, e1 interface{}, e2 interface{}, msgAndArgs ...interface{}) bool {
+	return compareTwoValues(t, e1, e2, []int {less, equal}, "\"%s\" is not less or equal than \"%s\"", msgAndArgs);
+}
+
+
+func compareTwoValues(t TestingT, e1 interface{}, e2 interface{}, allowedComparesResults []int, failMessage string, msgAndArgs ...interface{}) bool {
```

Comment:
How about using a struct with three boolean fields for each comparsion result, instead of a slice? I guess this needs a benchmark, but iterating over a slice of 1 to 2 elements seems like an overkill.

Reply:
@nineinchnick . I thought about that. The compare function returns constants `greater`, `equal`, `less`. Do you think iterating over the slice takes much time there? If we have a struct we should anyway interpret what the value we got. It will make the code more difficult.
I guess now it's readable and simple. What do you think?

Reply:
I think it's a great idea to extract the common logic for comparing two values. Do you think however that it would make more sense if it only focused on the comparison and the calling functions did the 'failMessage' formatting?

Reply:
@boyan-soubachov Thank you for your review! Oh, I guess I see what you mean: extracting raising final `Fail`  into upper level methods. Good input! I also thought about this idea. But I choose  the current one because:
 - The current approach minimize duplication of code (in the suggested solution we would need to copy paste code for call `Fail`. It's not much, but anyway...) 
 - We already have other checking for values `Kinds` and that values are comparable before. And I think it's another one plus to put calling `Fail` with `failMessage` inside this method.

Reply:
Fair enough, that makes sense